src/main/scripts/src/app/core/rest/project.service.ts

```diff
@@ -19,41 +19,93 @@
  */
 
 import {Injectable} from '@angular/core';
-import {HttpResponse} from '@angular/common/http';
-import {HttpClient} from '@angular/common/http';
+import {HttpClient, HttpResponse, HttpErrorResponse} from '@angular/common/http';
 
 import {Project} from '../dto/project';
 import {Observable} from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
+import {Permissions} from '../dto/permissions';
+import {Permission} from '../dto/permission';
+import {WorkspaceService} from '../workspace.service';
+import {ErrorObservable} from 'rxjs/observable/ErrorObservable';
+import {BadInputError} from '../error/bad-input.error';
+import {LumeerError} from '../error/lumeer.error';
 
 @Injectable()
 export class ProjectService {
 
-  constructor(private httpClient: HttpClient) {
+  constructor(private httpClient: HttpClient,
+    private workspaceService: WorkspaceService) {
   }
 
   public getProjects(orgCode: string): Observable<Project[]> {
-    return this.httpClient.get<Project[]>(ProjectService.apiPrefix(orgCode));
+    return this.httpClient.get<Project[]>(this.apiPrefix(orgCode));
   }
 
   public getProject(orgCode: string, projCode: string): Observable<Project> {
-    return this.httpClient.get<Project>(ProjectService.apiPrefix(orgCode, projCode));
+    return this.httpClient.get<Project>(this.apiPrefix(orgCode, projCode));
   }
 
   public deleteProject(orgCode: string, projCode: string): Observable<HttpResponse<object>> {
-    return this.httpClient.delete(ProjectService.apiPrefix(orgCode, projCode), {observe: 'response'});
+    return this.httpClient.delete(this.apiPrefix(orgCode, projCode), {observe: 'response'});
   }
 
   public createProject(orgCode: string, project: Project): Observable<HttpResponse<object>> {
-    return this.httpClient.post(ProjectService.apiPrefix(orgCode), project, {observe: 'response'});
+    return this.httpClient.post(this.apiPrefix(orgCode), project, {observe: 'response'});
   }
 
   public editProject(orgCode: string, projCode: string, project: Project): Observable<HttpResponse<object>> {
-    return this.httpClient.put(ProjectService.apiPrefix(orgCode, projCode), project, {observe: 'response'});
+    return this.httpClient.put(this.apiPrefix(orgCode, projCode), project, {observe: 'response'});
   }
 
-  private static apiPrefix(orgCode: string, projCode?: string): string {
+  public getPermissions(): Observable<Permissions> {
+    let orgCode = this.workspaceService.actualOrganizationCode;
```

Comment:
You can refactor all these duplicit `let code = this.workspaceService.actualOrganizationCode; let projCode = this.workspaceService.actualProjectCode;` to a `actualApiPrefix` method.